### ISSUE TITLE
Declaring an error using the shortcut declaration (for 1 error) results in a Segmentation Fault

### ISSUE THREAD

**Issue by hdorio** [ziglang/zig#9972]
### Remember to search before filing a new report

- [X] I searched for this bug and did not find it in the issue tracker, and furthermore, the title I used above will make this new bug report turn up in the search results for my query.

### Zig Version

0.9.0-dev.1352+6d6cf5984

### Steps to Reproduce

Launch the following examples with `zig test`

```zig
// file segfault1.zig
// $ zig test segfault1.zig
const Err: type = error.Hello;
test {
    const foo = Err;
    foo catch {};
}
//Segmentation fault
```

```zig
// file segfault2.zig
// $ zig test segfault2.zig
test {
    const Type: type = struct {
        err_field: error.Hello, // Segmentation fault
        // err_field: error{Hello} = undefined, // no segmentation fault
    };

    const foo = Type{};
    _ = foo;
}
//Segmentation fault
```

### Expected Behavior

A compiler error like the following when the declaration is inside the test block

```zig
// file error.zig
// $ zig test error.zig
test {
    const Err: type = error.Hello;
    const foo = Err;
    foo catch {};
}
//:2:28: error: expected type 'type', found 'error{Hello}'
//    const Err: type = error.Hello;
//                           ^
```

### Actual Behavior

When I assign an new error set with the shortcut `error.OneValue` on a `type` constant
```
const err: type = error.OneValue
```
Which does not make sense because as I unsderstand an error is not a type and the shortcut should result in something like
```
const err: error {OneValue} = (error {OneValue}).OneValue
```
Feel free to correct me if I'm wrong.

This results in a Segmentation Fault when running `zig run` or `zig test`
(only when the error declaration is outside the test block, see segfault1.zig)

Also when a struct field is declared as an `error` value with the shortcut this results again in a Segmentation fault.
```
struct { err: error.OneValue }
```

**Comment by RetroDev256** [comment]
duplicate of #8902

**Comment by hdorio** [comment]
fixed in [0.10.0](https://github.com/ziglang/zig/milestone/16)

```
$ ./zig-linux-x86_64-0.10.0/zig test segfault1.zig 
segfault1.zig:3:19: error: expected type 'type', found 'error{Hello}'
const Err: type = error.Hello;
                  ^~~~~~~~~~~
referenced by:
    test_0: segfault1.zig:5:17
    remaining reference traces hidden; use '-freference-trace' to see all reference traces

$ ./zig-linux-x86_64-0.10.0/zig test segfault2.zig 
segfault2.zig:4:24: error: expected type 'type', found 'error{Hello}'
    const Type: type = struct {
                       ^~~~~~
referenced by:
    test_0: segfault2.zig:4:24
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```
